src/playlist-selectors.js

```diff
@@ -103,91 +120,118 @@ export const comparePlaylistResolution = function(left, right) {
 
   rightWidth = rightWidth || window.Number.MAX_VALUE;
 
+  const leftBandwidth = getBandwidth(left, useAverageBandwidth);
+  const rightBandwidth = getBandwidth(right, useAverageBandwidth);
+
   // NOTE - Fallback to bandwidth sort as appropriate in cases where multiple renditions
-  // have the same media dimensions/ resolution
-  if (leftWidth === rightWidth &&
-      left.attributes.BANDWIDTH &&
-      right.attributes.BANDWIDTH) {
-    return left.attributes.BANDWIDTH - right.attributes.BANDWIDTH;
+  // have the same media dimensions/ resolution.
+  if (leftWidth === rightWidth && leftBandwidth && rightBandwidth) {
+    return leftBandwidth - rightBandwidth;
   }
   return leftWidth - rightWidth;
 };
 
-const simpleSelector = function(master, bandwidth, width, height) {
+const simpleSelector = function(master,
+                                bandwidth,
+                                width,
+                                height,
+                                useAverageBandwidth = false) {
   let sortedPlaylists = master.playlists.slice();
-  let bandwidthPlaylists = [];
-  let bandwidthBestVariant;
-  let resolutionPlusOne;
-  let resolutionBestVariant;
-  let haveResolution;
-  let resolutionPlusOneList = [];
-  let resolutionPlusOneSmallest = [];
-  let resolutionBestVariantList = [];
 
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+  stableSort(sortedPlaylists, comparePlaylistBandwidth.bind(null, useAverageBandwidth));
+
+  // convert the playlists to an intermediary representation to make comparisons easier
+  // and prevent us from re-determining bandwidth each time
+  let sortedPlaylistReps = sortedPlaylists.map((playlist) => {
```

Comment:
Do this first, and then use `stableSort` on this list for both `comparePlaylistBandwidth` and `comparePlaylistResolution`. This way both those functions do not need to add a new parameter `useAverageBandwidth` and you can use `comparePlaylistResolution` again instead of the anonymous function.  Then you just need to check `rep.bandwidth` in the compare functions (this makes `comparePlaylistBandwidth` a one-liner!). 

Id say the main argument against this approach which I predict will concern you, is that it changes the compare functions internally such that anyone using the exposed compare functions (even though we've never said you should do that) will break. Again I still believe this is ok. We've already broken overriding the compare functions but not the simple selector by moving these functions to their own module.